Reinstate customer subscription webhook handling

allouis opened the pull request:
🍦

allouis committed:
Revert "🐛 Fixed missing subscription attribution on free to paid upgrade (#21846)"

This reverts commit bd20ad3adb1f5400c5044b4fbed735648cd7ee27.

allouis committed:
WIP

9larsons committed:
Removed trailing commas

9larsons commented:
In reviewing this, it looks like:
1. Sag's original fix [commit](https://github.com/TryGhost/Ghost/commit/bd20ad3adb1f5400c5044b4fbed735648cd7ee27) was reverted.
2. Per the approach [mentioned](https://linear.app/ghost/issue/ONC-729/ghost-does-not-create-new-subscriptions-for-new-subscriptions-created#comment-2d262db7) in the issue, a new tag/event was added in the `CheckoutSessionEventService` to attempt updating attribution for `checkout.session.completed` events. 

Unit tests are failing in multiple places, so I suspect that's what's left to handle.

9larsons committed:
Fixed stripe unit tests

9larsons committed:
Added sample test

9larsons committed:
Added some debug logging, updated model to ignore updated_at, passed correct subscription id

9larsons committed:
Fixed e2e test

Note: we previously had no test coverage with the `checkout.session.completed` handler in the case of an existing member. This had to modify a couple pieces to allow for that.

Because we're editing the subscriptionCreatedEvent, the memberCreatedEvent info needs to be skipped as it may not match (which is expected).

9larsons committed:
Updated stripe package tests

9larsons committed:
Added and fixed event storage unit tests

9larsons committed:
Removed clog

9larsons committed:
Update ghost/member-events/lib/SubscriptionAttributionEvent.js

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

9larsons commented:
@allouis Needs your review. Idk how to take over a PR short of opening a new one, which maybe is best here given the noisy commit history.

9larsons commented:
Adding @cmraible for review.

9larsons commented:
Closing this to open a new PR. I've pushed the revert commit separately to get it out with the next release.
